### PR TITLE
Upgrades + Updates

### PR DESCRIPTION
**Pull request by stefanpenner**
* ember-cli
* ember-try
* syntax / es6
* test usage
* test execa usage (soon library itself)

### CODE REVIEW

**Comment by stefanpenner** [comment]
@rwjblue i was having two issues, I could use your eyes on:

* `ember-template-lint ` was picking up stuff in node_modules (seems strange ...)
* `eslint .` was getting confused by both mocha and qunit in this code-base, and my various configuration attempts to correct this failed. Could you share some of your magic eslint foo?

**Comment by stefanpenner** [comment]
- [ ] fix mocha tests

**Comment by stefanpenner** [comment]
Hmm, travis YUNO run?

File: .travis.yml [hunk 1]
```diff
@@ -1,42 +1,64 @@
 ---
 language: node_js
 node_js:
+  # we recommend testing addons with the same minimum supported node version as Ember CLI
+  # so that your addon works for all apps
   - "6"
-  - "8"
```
Comment (step2yeung) [inline .travis.yml]: We only want to test on the minimum and not all supported versions? Why is that?

Comment (stefanpenner) [reply .travis.yml]: To avoid consuming too many resources on travis.

**Comment by stefanpenner** [comment]
Not sure whats up with travis, going to reopen this PR.

**Pull request closed without merging**